diff -crB a/qpid/java/broker/bin/qpid-passwd b/qpid/java/broker/bin/qpid-passwd
*** a/qpid/java/broker/bin/qpid-passwd	2013-01-23 14:34:46.000000000 -0600
--- b/qpid/java/broker/bin/qpid-passwd	2013-01-23 14:33:48.000000000 -0600
***************
*** 24,30 ****
  fi
  
  # Set classpath to include Qpid jar with all required jars in manifest
! QPID_LIBS=$QPID_HOME/lib/qpid-all.jar
  
  # Set other variables used by the qpid-run script before calling
  export JAVA=java \
--- 24,30 ----
  fi
  
  # Set classpath to include Qpid jar with all required jars in manifest
! QPID_LIBS="$QPID_HOME/lib/qpid-all.jar:$QPID_HOME/lib/opt/*:$QPID_HOME/lib/opt/qpid-deps/*"
  
  # Set other variables used by the qpid-run script before calling
  export JAVA=java \
diff -crB a/qpid/java/broker/bin/qpid-server b/qpid/java/broker/bin/qpid-server
*** a/qpid/java/broker/bin/qpid-server	2013-01-23 14:34:46.000000000 -0600
--- b/qpid/java/broker/bin/qpid-server	2013-01-23 14:33:48.000000000 -0600
***************
*** 23,30 ****
  fi
  
  if [ -z "$QPID_WORK" ]; then
!     echo "Setting QPID_WORK to $HOME as default"
!     QPID_WORK=$HOME
  fi
  
  # Set to help us get round the manifold problems of ps/pgrep on various
--- 23,29 ----
  fi
  
  if [ -z "$QPID_WORK" ]; then
!     export QPID_WORK=$QPID_HOME
  fi
  
  # Set to help us get round the manifold problems of ps/pgrep on various
***************
*** 34,41 ****
  fi
  
  # Set classpath to include the qpid-all manifest jar, and any jars supplied in lib/opt
! QPID_LIBS="$QPID_HOME/lib/qpid-all.jar:$QPID_HOME/lib/opt/*"
  
  # Set other variables used by the qpid-run script before calling
! export JAVA=java \
         JAVA_VM=-server \
--- 33,40 ----
  fi
  
  # Set classpath to include the qpid-all manifest jar, and any jars supplied in lib/opt
! QPID_LIBS="$QPID_HOME/lib/qpid-all.jar:$QPID_HOME/lib/opt/*:$QPID_HOME/lib/opt/qpid-deps/*"
  
  # Set other variables used by the qpid-run script before calling
! export JAVA=/awips2/java/bin/java \
         JAVA_VM=-server \
diff -crB a/qpid/java/build.deps b/qpid/java/build.deps
*** a/qpid/java/build.deps	2013-01-23 13:54:15.000000000 -0600
--- b/qpid/java/build.deps	2013-01-23 13:52:28.000000000 -0600
***************
*** 19,29 ****
  
! commons-beanutils-core=lib/required/commons-beanutils-core-1.8.3.jar
  commons-cli=lib/required/commons-cli-1.2.jar
! commons-codec=lib/required/commons-codec-1.6.jar
  commons-collections=lib/required/commons-collections-3.2.1.jar
! commons-configuration=lib/required/commons-configuration-1.8.jar
  commons-digester=lib/required/commons-digester-1.8.1.jar
! commons-lang=lib/required/commons-lang-2.6.jar
  commons-logging=lib/required/commons-logging-1.1.1.jar
  
  derby-db=lib/required/derby-10.8.2.2.jar
--- 19,29 ----
  
! commons-beanutils-core=lib/required/commons-beanutils-1.8.3.jar
  commons-cli=lib/required/commons-cli-1.2.jar
! commons-codec=lib/required/commons-codec-1.4.jar
  commons-collections=lib/required/commons-collections-3.2.1.jar
! commons-configuration=lib/required/commons-configuration-1.6.jar
  commons-digester=lib/required/commons-digester-1.8.1.jar
! commons-lang=lib/required/commons-lang-2.3.jar
  commons-logging=lib/required/commons-logging-1.1.1.jar
  
  derby-db=lib/required/derby-10.8.2.2.jar
***************
*** 35,38 ****
  geronimo-openejb=lib/required/geronimo-ejb_3.0_spec-1.0.1.jar
! geronimo-servlet=lib/required/geronimo-servlet_2.5_spec-1.2.jar

  junit=lib/required/junit-3.8.1.jar
--- 35,38 ----
  geronimo-openejb=lib/required/geronimo-ejb_3.0_spec-1.0.1.jar
! geronimo-servlet=lib/required/geronimo-servlet_2.5_spec-1.1.2.jar

  junit=lib/required/junit-3.8.1.jar
***************
*** 45,51 ****
  slf4j-api=lib/required/slf4j-api-1.6.4.jar
  slf4j-log4j=lib/required/slf4j-log4j12-1.6.4.jar
  
! xalan=lib/required/xalan-2.7.0.jar
  
 jetty=lib/required/jetty-server-7.6.3.v20120416.jar
 jetty-continuation=lib/required/jetty-continuation-7.6.3.v20120416.jar
--- 45,51 ----
  slf4j-api=lib/required/slf4j-api-1.6.4.jar
  slf4j-log4j=lib/required/slf4j-log4j12-1.6.4.jar
  
! xalan=lib/required/xalan-2.7.1.jar
  
 jetty=lib/required/jetty-server-7.6.3.v20120416.jar
 jetty-continuation=lib/required/jetty-continuation-7.6.3.v20120416.jar
***************
*** 63,70 ****
  
  felix.libs=${felix-main}
  
! jackson-core=lib/required/jackson-core-asl-1.9.0.jar
! jackson-mapper=lib/required/jackson-mapper-asl-1.9.0.jar
  
  commons-configuration.libs = ${commons-beanutils-core} ${commons-digester} \
    ${commons-codec} ${commons-lang} ${commons-collections} ${commons-configuration}
--- 63,70 ----
  
  felix.libs=${felix-main}
  
! jackson-core=lib/required/jackson-core-asl-1.7.3.jar
! jackson-mapper=lib/required/jackson-mapper-asl-1.7.3.jar
  
  commons-configuration.libs = ${commons-beanutils-core} ${commons-digester} \
    ${commons-codec} ${commons-lang} ${commons-collections} ${commons-configuration}
diff -crB a/qpid/java/common/bin/qpid-jaddr b/qpid/java/common/bin/qpid-jaddr
*** a/qpid/java/common/bin/qpid-jaddr	2013-01-23 14:41:58.000000000 -0600
--- b/qpid/java/common/bin/qpid-jaddr	2013-01-23 14:40:35.000000000 -0600
***************
*** 24,30 ****
  fi
  
  # Set classpath to include Qpid jar with all required jars in manifest
! QPID_LIBS=$QPID_HOME/lib/qpid-all.jar
  
  # Set other variables used by the qpid-run script before calling
  export JAVA=java \
--- 24,30 ----
  fi
  
  # Set classpath to include Qpid jar with all required jars in manifest
! QPID_LIBS="$QPID_HOME/lib/qpid-all.jar:$QPID_HOME/lib/opt/*:$QPID_HOME/lib/opt/qpid-deps/*"
  
  # Set other variables used by the qpid-run script before calling
  export JAVA=java \
diff -crB a/qpid/java/common/bin/qpid-run b/qpid/java/common/bin/qpid-run
*** a/qpid/java/common/bin/qpid-run	2013-01-23 14:41:58.000000000 -0600
--- b/qpid/java/common/bin/qpid-run	2013-01-23 14:40:35.000000000 -0600
***************
*** 62,69 ****
  fi
  
  if [ -z "$QPID_WORK" ]; then
!     log $INFO Setting QPID_WORK to $HOME as default
!     QPID_WORK=$HOME
  fi
  
  if [ -z "$JAVA" ]; then
--- 62,68 ----
  fi
  
  if [ -z "$QPID_WORK" ]; then
!     QPID_WORK=$QPID_HOME
  fi
  
  if [ -z "$JAVA" ]; then
